commit 23b06aafec4939854827697e2c68d8dfdcbd1c07
Author: Michael Still <mikal@stillhq.com>

    Remove mypy union line which breaks older pythons.

    Change I084336ba41147f824b92dc07235e5f19b7ac4a9c introduced mypy
    syntax which breaks Python releases before 3.10. Unfortunately, for
    2024.1 we commit to supporting Python back to 3.8.

    Specifically, you receive this error message if you run pep8:

    pep8 mypy.....................................................................Failed
    pep8 - hook id: mypy
    pep8 - exit code: 1
    pep8
    pep8 openstack/object_store/v1/_proxy.py: note: In member "generate_temp_url" of class "Proxy":
    pep8 openstack/object_store/v1/_proxy.py:1049:21: error: X | Y syntax for unions requires Python 3.10  [syntax]
    pep8 Found 1 error in 1 file (checked 410 source files)

    I asked some buddies, and we're fairly sure that this line would
    crash the runtime on Python 3.8, because its a syntax error.

    So instead, let's remove this line until we've dropped support for
    Python versions before 3.10.

diff --git a/openstack/object_store/v1/_proxy.py b/openstack/object_store/v1/_proxy.py
index a63d9dc2..0d4b6ab7 100644
--- a/openstack/object_store/v1/_proxy.py
+++ b/openstack/object_store/v1/_proxy.py
@@ -1046,7 +1046,6 @@ class Proxy(proxy.Proxy):
                 method.upper(),
             )

-        expiration: float | int
         if not absolute:
             expiration = _get_expiration(timestamp)
         else:
